install-steps.md

```diff
@@ -1,193 +1,821 @@
+
+~~~sh
+<OUTPUT_OMITTED>
+dhcp-host=<NIC2-mac-address>,openshift-worker-1.openshift.example.com,<ip-of-worker-1>
+dhcp-host=<NIC2-mac-address>,openshift-worker-2.openshift.example.com,<ip-of-worker-2>
+~~~
+
+NOTE: Remove the provisioner host entry as it is replaced by the openshift-worker-2
+
+Once the change as been added and saved, restart the `dnsmasq` service
+~~~sh
+systemctl restart dnsmasq
+~~~
+
+## Deploy the Provisioner node as a Worker Node using Metal3
+
+In order to convert the provisioner node as a worker node ensure the following:
+
+1. `ssh` to a system that can access the out of band management network of the current provisioner node
+2. Poweroff the node using the `ipmitool` and confirm the provisioning node is powered off
+
+    ~~~sh
+    ssh <server-with-access-to-management-net>
+    # Use the user, password and Management net IP adddress to shutdown the system
+    ipmitool -I lanplus -U <user> -P <password> -H <management-server-ip> -p 6230 power off
+    # Confirm the server is powered down
+    ipmitool -I lanplus -U <user> -P <password> -H <management-server-ip> -p 6230 power status
+    Chassis Power is off
+    ~~~
+3. Get `base64` strings for the Out of band Management credentials. In the example user is `root`, password is `calvin`
+
+    ~~~sh
+    # Use echo -ne, otherwise you will get your secrets with \n which will cause issues
+    # Get root username in base64
+    echo -ne "root" | base64
+    # Get root password in base64
+    echo -ne "calvin" | base64
+    ~~~
+4. Configure the BaremetalHost `bmh.yaml`
+
+    ~~~yaml
+    ---
+    apiVersion: v1
+    kind: Secret
+    metadata:
+      name: openshift-worker-2-bmc-secret
+    type: Opaque
+    data:
+      username: ca2vdAo=
+      password: MWAwTWdtdC0K
+    ---
+    apiVersion: metal3.io/v1alpha1
+    kind: BareMetalHost
+    metadata:
+      name: openshift-worker-2
+    spec:
+      online: true
+      bootMACAddress: <NIC1-mac-address>
+      bmc:
+        address: ipmi://<   out-of-band-ip> 
+        credentialsName: openshift-worker-2-bmc-secret
+    ~~~
+    
+5. Create the BaremetalHost
+
+    ~~~sh
+    oc -n openshift-machine-api create -f bmh.yaml
+    ~~~
+6. The Node will be powered up and inspected
+
+    ~~~sh
+    oc -n openshift-machine-api get bmh openshift-worker-2
+
+    NAME                 STATUS   PROVISIONING STATUS   CONSUMER   BMC                 HARDWARE PROFILE   ONLINE   ERROR
+    openshift-worker-2   OK       inspecting                       ipmi://<out-of-band-ip>                      true     
+    ~~~
+7. Once the inspection finishes the node will be ready to be provisioned
+
+    ~~~sh
+    oc -n openshift-machine-api get bmh openshift-worker-2
+
+    NAME                 STATUS   PROVISIONING STATUS   CONSUMER   BMC                 HARDWARE PROFILE   ONLINE   ERROR
+    openshift-worker-2   OK       ready                            ipmi://<out-of-band-ip>   unknown            true     
```

Comment:
I'd say to add instructions on how to see the ironic stuff under the hood:

* Create a clouds.yaml file with the following content:
```
clouds:
  metal3-bootstrap:
    auth_type: none
    baremetal_endpoint_override: http://172.22.0.2:6385
    baremetal_introspection_endpoint_override: http://172.22.0.2:5050
  metal3:
    auth_type: none
    baremetal_endpoint_override: http://172.22.0.3:6385
    baremetal_introspection_endpoint_override: http://172.22.0.3:5050
```

* Install the openstack and ironic cli (I prefer to set a virtualenv and pip on that venv)
* Then query ironic as:
```
export OS_CLOUD="metal3"
openstack baremetal node list
```

Reply:
What about running openstack client from a container with podman?

Reply:
> What about running openstack client from a container with podman?

It is a bit tricky because AFAIK the openstack-client rpm is only available in the OSP channels so I'm not sure if we provide an openstack-client container. That being said, it is similar in terms of (in)security to install the tool via pip than pulling an insecure container image :D

Reply:
I know Steve Hardy had something going here regarding osp client in a container: https://github.com/openshift/ironic-image/pull/17

Reply:
Does this piece really belong in the installation doc? Should this be a different doc regarding troubleshooting? @e-minguez @karmab

Reply:
it can go in a troubleshooting doc indeed